scripts/github_issue_retriever.py

```diff
@@ -5,16 +5,50 @@
 
 import argparse
 from logging import getLogger
+from typing import List, Tuple
 
 from bugbug import db, github
-from bugbug.utils import zstd_compress
+from bugbug.github import IssueDict
+from bugbug.utils import extract_private, zstd_compress
 
 logger = getLogger(__name__)
 
 
+def replace_with_private(
+    original_data: List[IssueDict],
+) -> Tuple[List[IssueDict], List[IssueDict], set]:
+    """Replace title and body of automatically closed public issues.
+
+    Replace them with title and body of a corresponding private issue
+    to account for moderation workflow in webcompat repository
+    """
+    updated_ids = set()
+    updated_issues = []
+    for item in original_data:
+        if item["title"] == "Issue closed.":
+            extracted = extract_private(item["body"])
+            if extracted is None:
+                continue
+
+            owner, repo, issue_number = extracted
+            private_issue = github.fetch_issue_by_number(owner, repo, issue_number)
+            if private_issue:
+                item["title"] = private_issue["title"]
+                item["body"] = private_issue["body"]
+                updated_ids.add(item["id"])
+                updated_issues.append(item)
+
+    return original_data, updated_issues, updated_ids
```

Comment:
Sorry, one last nit: since the `original_data` dict is not changed (only the elements inside it are), you can avoid returning it from the function.

Reply:
Ah right, thanks :)